movie-mlops-project/preprocessing/preprocessing.py

```
@@ -0,0 +1,421 @@
+            # 주요 장르 원핫 인코딩
+            major_genres = {
+                28: 'Action', 12: 'Adventure', 16: 'Animation', 35: 'Comedy',
+                80: 'Crime', 99: 'Documentary', 18: 'Drama', 10751: 'Family',
+                14: 'Fantasy', 36: 'History', 27: 'Horror', 10402: 'Music',
+                9648: 'Mystery', 10749: 'Romance', 878: 'Science Fiction',
+                53: 'Thriller', 10752: 'War', 37: 'Western'
+            }
+            
+            for genre_id, genre_name in major_genres.items():
+                df[f'is_{genre_name.lower().replace(" ", "_")}'] = df['genre_ids'].apply(
+                    lambda x: 1 if genre_id in x else 0
```

Ruff (0.12.2)

Function definition does not bind loop variable `genre_id`

(B023)